tests/unit/models/keras/test_utils.py

```diff
@@ -102,3 +106,28 @@ def test_sample_with_replacement_seems_correct(rank: int) -> None:
     x = tf.cast(x[:, 0], dtype=tf.float32)
     assert (tf.reduce_mean(mean) - tf.reduce_mean(x)) < 1
     assert tf.math.abs(tf.math.reduce_std(mean) - tf.math.reduce_std(x) / 10.0) < 0.1
+
+
+@pytest.mark.parametrize("size", [2, 10])
+@pytest.mark.parametrize("num_samples", [0, 1, 10])
+def test_sample_model_index_call_shape(size: int, num_samples: int) -> None:
+    indices = sample_model_index(size, num_samples)
+
+    assert indices.shape == (num_samples,)
```

Comment:
You could probably add some more tests for sample_model_index (e.g. check that the first size indices are unique)